benchmarktests/target_secret_totp.go

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package benchmarktests
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"math/rand"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/vault/api"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+const (
+	TOTPReadTestType    = "totp_read"
+	TOTPWriteTestType   = "totp_write"
+	TOTPReadTestMethod  = "GET"
+	TOTPWriteTestMethod = "POST"
+)
+
+func init() {
+	TestList[TOTPReadTestType] = func() BenchmarkBuilder { return &TOTPTest{action: "read"} }
+	TestList[TOTPWriteTestType] = func() BenchmarkBuilder { return &TOTPTest{action: "write"} }
+}
+
+type TOTPTest struct {
+	pathPrefix string
+	header     http.Header
+	config     *TOTPSecretTestConfig
+	numkeys    int
+	keysize    int
+	action     string
+	logger     hclog.Logger
+}
+
+type TOTPSecretTestConfig struct {
+	NumKeys int `hcl:"numkeys,optional"`
+	KeySize int `hcl:"keysize,optional"`
+}
```

Comment:
In general we want to expose all possible API endpoint parameters to allow flexibility in testing configuration. Can we add the parameters listed in the docs? https://developer.hashicorp.com/vault/api-docs/secret/totp#create-key